lib/units/auth/ldap.js

```diff
@@ -57,65 +59,79 @@ export default (function(options) {
         res.sendFile(pathutil.reactFrontend('dist/auth/auth-ldap.html'))
     })
     app.post('/auth/api/v1/ldap', function(req, res) {
-        var log = logger.createLogger('auth-ldap')
+        const log = logger.createLogger('auth-ldap')
         log.setLocalIdentifier(req.ip)
-        switch (req.accepts(['json'])) {
-        case 'json':
-            requtil.validate(req, function() {
-                req.checkBody('username').notEmpty()
-                req.checkBody('password').notEmpty()
+        if (req.accepts(['json']) !== 'json') {
+            res.send(406)
+            return
+        }
+
+        requtil.validate(req, function() {
+            req.checkBody('username').notEmpty()
+            req.checkBody('password').notEmpty()
+        })
+            .then(function() {
+                return ldaputil.login(options.ldap, req.body.username, req.body.password)
             })
-                .then(function() {
-                    return ldaputil.login(options.ldap, req.body.username, req.body.password)
+            .then(function(user) {
+                const privilege = ldaputil.determinePrivilege(user, options.ldap.privilegeMapping)
+
+                const username = user[options.ldap.username.field]
+                const usernameValue = Array.isArray(username) ? username[0] : username
+
+                const userEmail = ldaputil.email(user)
+
+                log.info('Authenticated "%s" with privilege "%s"', userEmail, privilege)
+
+                const token = jwtutil.encode({
+                    payload: {
+                        email: userEmail
+                        , name: usernameValue
+                    }
+                    , secret: options.secret
+                    , header: {
+                        exp: Date.now() + ONE_DAY
+                    }
                 })
-                .then(function(user) {
-                    log.info('Authenticated "%s"', ldaputil.email(user))
-                    var token = jwtutil.encode({
-                        payload: {
-                            email: ldaputil.email(user)
-                            , name: user[options.ldap.username.field]
-                        }
-                        , secret: options.secret
-                        , header: {
-                            exp: Date.now() + ONE_DAY
-                        }
-                    })
+
+                return dbapi.saveUserAfterLogin({
```

Comment:
Auth units should not use the database for consistency. It's currently used to pass the contact info in `/auth/contact`, but it's not optimal in it's current implementation. Let's not use it elsewhere for now